### PR TITLE
[analytics] Track Projects & Teams

### PR DESCRIPTION
**Pull request by jakobhero**
## Description

Implements tracking for the dashboard

## How to test

- Go to Segment's staging untrusted source
- This PR should generate page and tracking events a user navigating through the dashboard

## Release Notes

```release-note
NONE
```

/werft analytics=segment|TEZnsG4QbLSxLfHfNieLYGF4cDwyFWoe
/werft with-clean-slate-deployment=true

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **geropl** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @geropl` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/dashboard/src/projects/NewProject.tsx
Comment (AlexTugarev) [inline components/dashboard/src/projects/NewProject.tsx:119]: heads up, that organization might be already authorized and we've just updated the selection of repos.

File: components/dashboard/src/projects/NewProject.tsx
Comment (AlexTugarev) [inline components/dashboard/src/projects/NewProject.tsx:111]: nit: we don't know if the intention is to add an org here. this is more generic, something like "show a github page and let me decide what to do"

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **csweichel** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @csweichel` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**
- **[components/blobserve/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/blobserve/OWNERS)**
- **[components/common-go/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/common-go/OWNERS)**
- **[components/content-service/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/content-service/OWNERS)**
- **[components/content-service-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/content-service-api/OWNERS)**
- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**
- **[components/ee/agent-smith/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ee/agent-smith/OWNERS)**
- **[components/ee/db-sync/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ee/db-sync/OWNERS)**
- **[components/ee/kedge/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ee/kedge/OWNERS)**
- **[components/ee/payment-endpoint/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ee/payment-endpoint/OWNERS)**
- **[components/ee/ws-scheduler/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ee/ws-scheduler/OWNERS)**
- **[components/gitpod-cli/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-cli/OWNERS)**
- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/image-builder/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder/OWNERS)**
- **[components/image-builder-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-api/OWNERS)**
- **[components/local-app/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/local-app/OWNERS)**
- **[components/local-app-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/local-app-api/OWNERS)**
- **[components/registry-facade/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/registry-facade/OWNERS)**
- **[components/registry-facade-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/registry-facade-api/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- **[components/service-waiter/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/service-waiter/OWNERS)**
- **[components/supervisor/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/OWNERS)**
- **[components/supervisor-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor-api/OWNERS)**
- **[components/workspacekit/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/workspacekit/OWNERS)**
- **[components/ws-daemon/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-daemon/OWNERS)**
- **[components/ws-daemon-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-daemon-api/OWNERS)**
- **[components/ws-manager/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager/OWNERS)**
- **[components/ws-manager-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-api/OWNERS)**
- **[components/ws-manager-bridge/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-bridge/OWNERS)**
- **[components/ws-manager-bridge-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-bridge-api/OWNERS)**
- **[components/ws-proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-proxy/OWNERS)**
- **[dev/gpctl/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/dev/gpctl/OWNERS)**
- **[dev/loadgen/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/dev/loadgen/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/dashboard/src/projects/NewProject.tsx
Comment (jakobhero) [reply components/dashboard/src/projects/NewProject.tsx]: thanks, it made sense to move it to where the 'Add GitHub Orgs or Account' button is actually pushed in `/new`

File: components/dashboard/src/projects/NewProject.tsx
Comment (jakobhero) [reply components/dashboard/src/projects/NewProject.tsx:113]: thanks for the heads up, keeping it this way for now due to the lack of a better name.

File: components/dashboard/src/projects/Projects.tsx
Comment (gtsiolis) [inline components/dashboard/src/projects/Projects.tsx:92]: **suggestion**: There's also an _Invite Members_ button on the Projects list. Shall we also track this?

File: components/dashboard/src/teams/Members.tsx
Comment (gtsiolis) [inline components/dashboard/src/teams/Members.tsx:214]: **question**: Since we'll track the button that triggers this modal, does it help to track if the modal opened?

Comment (gtsiolis) [inline components/dashboard/src/teams/Members.tsx:214]: **thought**: Instead of tracking this shall we track the [Copy Invite URL](https://github.com/gitpod-io/gitpod/pull/5168/files#diff-53ff0bdfe65ca0d0e6a4af4ac305ab6a4f94c852f14e509b915659e8d653c820R229) below?

**Review by gtsiolis — COMMENTED** [review]
Great to see this coming into life @jakobhero! 🍊 

I was in the neighhood and left some minor comments below. 🚶

File: components/dashboard/src/teams/Members.tsx
Comment (jankeromnes) [inline components/dashboard/src/teams/Members.tsx:214]: Note: `onEnter` is only triggered if you already have the modal open, and you hit the "Enter" key (e.g. if you do a lot of keyboard navigation). So this is likely not the correct event handler to track this modal. Instead, you should probably track "clicks" on the "Copy Invite URL" button as @gtsiolis suggested. 🙂

File: components/dashboard/src/projects/Projects.tsx
Comment (jakobhero) [reply components/dashboard/src/projects/Projects.tsx]: Done!👍🏼

File: components/dashboard/src/teams/Members.tsx
Comment (jakobhero) [reply components/dashboard/src/teams/Members.tsx]: @gtsiolis Generally, I am tracking two different things: 1) What buttons were clicked and where they are positioned (e.g. `dashboard_button_clicked`) and 2) the effects on the creation / alteration / deletion on other entities the user interaction has had. In this instance, I want to track the specific `inviteUrl` as this will allow to match it up with users who follow the invitation. The URL is not a general property of a button though, so I put it in a dedicated event. As the pressing of the button will inevitably cause the URL to be displayed, I moved the call to track the request for the `inviteUrl` to the onClick event that will open the modal and removed the `dashboard_button_clicked` event from there.

Comment (jakobhero) [reply components/dashboard/src/teams/Members.tsx]: @jankeromnes thanks for the explanation here! I was thinking about tracking the Copy Invite URL button too, but this would for situations in which users copy the URL manually and invite a new member to the team without us having visibility of  the user sending an invite (or, more specifically, demanding an invite URL) on the analytics end as the event is only caused when the button is pressed.

File: components/dashboard/src/Analytics.tsx
Comment (jakobhero) [reply components/dashboard/src/Analytics.tsx]: oh great, thank you! is it still going to be the same field name though (`x-real-ip`)?

File: components/server/src/workspace/gitpod-server-impl.ts
Comment (jakobhero) [reply components/server/src/workspace/gitpod-server-impl.ts]: as discussed on friday, only signed up users are tracked in the latest change and anonymous users are never tracked. if a user has to log in first, the initially requested URL and referrer are stored until the user is logged in, at which point one `trackLocation` call is made with the initial URL and referrer.

File: components/dashboard/src/Analytics.tsx
Comment (jakobhero) [reply components/dashboard/src/Analytics.tsx]: thanks, modified it accordingly!

File: components/dashboard/src/App.tsx
Comment (jakobhero) [reply components/dashboard/src/App.tsx]: > Other than that, we need to have a go on the anonymous tracking and a rebase. Ah, there are also some merge conflicts to solve and the configuration page was still the old one.

anonymous tracking is now solved and the branch has been rebased. Also, I included the tracking for the new workspace button in the new configuration page.

**Comment by roboquat** [comment]
@jakobhero: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by AlexTugarev** [comment]
@jakobhero, could you update again, please? Let's merge asap.

Also release notes are blocking. I think it'd fine to proceed without an entry there. Your call.

**Comment by svenefftinge** [comment]
I like the generic click tracker much better. Could you rebase and squash all commits? Also please minimize the changes (there are so many unrelated formatting changes, etc.)

**Review by svenefftinge — APPROVED** [review]
Looks great! Could you remove the formatting changes and rebase?

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5168#pullrequestreview-758605524" title="Approved">svenefftinge</a>*

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [svenefftinge]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [svenefftinge]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [svenefftinge]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Review by svenefftinge — APPROVED** [review]
LGTM

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 11921f30961be2649dfcc04d06ce2c80db5a0a4f</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5168#pullrequestreview-760655729" title="Approved">svenefftinge</a>*

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [svenefftinge]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [svenefftinge]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [svenefftinge]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5168#issuecomment-924737381" title="Approved">svenefftinge</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/5168#issuecomment-924737381" title="Approved">svenefftinge</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [svenefftinge]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [svenefftinge]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [svenefftinge]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by svenefftinge** [comment]
🎉

**Comment by roboquat** [comment]
@gtsiolis: [dog image](https://random.dog/65bf4e16-3d69-4115-b116-f1bda1b9de6b.jpg)

<details>

In response to [this](https://github.com/gitpod-io/gitpod/pull/5168#issuecomment-924751150):

>/woof


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Pull request merged**
